vcr/oidc4vci/types.go

```diff
@@ -0,0 +1,117 @@
+
+// ProviderMetadataWellKnownPath defines the well-known path for retrieving OpenID ProviderMetadata
+// Specified by https://www.rfc-editor.org/rfc/rfc8414.html#section-3
+const ProviderMetadataWellKnownPath = "/.well-known/oauth-authorization-server"
+
+// CredentialIssuerMetadataWellKnownPath defines the well-known path for retrieving OIDC4VCI CredentialIssuerMetadata
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-issuer-metadata-
+const CredentialIssuerMetadataWellKnownPath = "/.well-known/openid-credential-issuer"
+
+// VerifiableCredentialJSONLDFormat defines the JSON-LD format identifier for Verifiable Credentials.
+const VerifiableCredentialJSONLDFormat = "ldp_vc"
+
+// CredentialIssuerMetadata defines the OIDC4VCI Credential Issuer Metadata.
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-issuer-metadata
+type CredentialIssuerMetadata struct {
+	// CredentialIssuer defines the identifier of the credential issuer.
+	CredentialIssuer string `json:"credential_issuer"`
+
+	// CredentialEndpoint defines where the wallet can send a request to retrieve a credential.
+	CredentialEndpoint string `json:"credential_endpoint"`
+
+	// CredentialsSupported defines metadata about which credential types the credential issuer can issue.
+	CredentialsSupported []map[string]interface{} `json:"credentials_supported"`
+}
+
+// OAuth2ClientMetadata defines the OAuth2 Client Metadata, extended with OIDC4VCI parameters.
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-client-metadata.
+type OAuth2ClientMetadata struct {
+	// CredentialOfferEndpoint defines URL of the verifiable credential wallet's offer endpoint
+	CredentialOfferEndpoint string `json:"credential_offer_endpoint"`
+}
+
+// ProviderMetadata defines the OpenID Connect Provider metadata.
+// Specified by https://www.rfc-editor.org/rfc/rfc8414.txt
+type ProviderMetadata struct {
+	// Issuer defines the authorization server's identifier, which is a URL that uses the "https" scheme and has no query or fragment components.
+	Issuer string `json:"issuer"`
+
+	// TokenEndpoint defines the URL of the authorization server's token endpoint [RFC6749].
+	TokenEndpoint string `json:"token_endpoint"`
+
+	// PreAuthorizedGrantAnonymousAccessSupported indicates whether anonymous access (requests without client_id)
+	// for pre-authorized code grant flows.
+	// See https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-oauth-20-authorization-serv
+	PreAuthorizedGrantAnonymousAccessSupported bool `json:"pre-authorized_grant_anonymous_access_supported"`
+}
+
+// CredentialOffer defines credentials offered by the issuer to the wallet.
+// Specified by https://openid.net/specs/openid-4-verifiable-credential-issuance-1_0.html#name-credential-offer-parameters
+type CredentialOffer struct {
```

Comment:
Any info on what is to be expected under `interface{}`

Reply:
Depends on the format (jwt/json ld vc, etc), so see spec, I'd say.